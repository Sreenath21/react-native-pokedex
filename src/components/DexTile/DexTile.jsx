import React from 'react';
import {View, Text} from 'react-native';

import HighLighter from '../Highlighter/HighLighter';
import styles from './DexTile.style';

const DexTile = ({attribute, value}) => {
  return (
    <View style={styles.container}>
      <Text style={styles.title}>{attribute}</Text>
      {typeof value === 'string' ? (
        <Text style={styles.value}>{value}</Text>
      ) : (
        <HighLighter textArray={value} />
      )}
    </View>
  );
};

export default DexTile;
