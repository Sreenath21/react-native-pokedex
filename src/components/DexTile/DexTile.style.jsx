import {StyleSheet} from 'react-native';

import colors from '../../utils/colors';

export default StyleSheet.create({
  container: {
    flex: 1,
    rowGap: 10,
  },
  title: {
    fontSize: 16,
    fontWeight: 'bold',
    color: colors.primary,
  },
  value: {
    fontSize: 16,
    color: colors.primary,
  },
});
