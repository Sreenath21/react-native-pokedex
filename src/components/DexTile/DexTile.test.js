import 'react-native';
import React from 'react';
import DexTile from './DexTile';
import renderer from 'react-test-renderer';

it('Checking DexTile component rendering correctly', () => {
  const tree = renderer.create(<DexTile attribute="abilities" value="Blaze" />);
  expect(tree).toMatchSnapshot();
});
