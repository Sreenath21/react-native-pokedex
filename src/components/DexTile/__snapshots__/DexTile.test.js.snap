// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking DexTile component rendering correctly 1`] = `
<View
  style={
    {
      "flex": 1,
      "rowGap": 10,
    }
  }
>
  <Text
    style={
      {
        "color": "#391363",
        "fontSize": 16,
        "fontWeight": "bold",
      }
    }
  >
    abilities
  </Text>
  <Text
    style={
      {
        "color": "#391363",
        "fontSize": 16,
      }
    }
  >
    Blaze
  </Text>
</View>
`;
